CONTRIBUTING.md

```diff
@@ -111,73 +143,80 @@ Make sure you document each commit and squash them appropriately. You can use th
-2. Get the source code.
-
-		$ git clone git://github.com/sbt/sbt.git
-		$ cd sbt
+### Branch to work against
 
-3. The default branch is the development branch [1.0.x](https://github.com/sbt/sbt/tree/1.0.x), which contains the latest code for the next major sbt release.  To build a specific release or commit, switch to the associated tag.  The tag for the latest stable release is [v0.13.13](https://github.com/sbt/sbt/tree/v0.13.13):
+sbt uses two branches for development:
 
-		$ git checkout v0.13.13
+- Development branch: `1.x` (this is also called "master")
+- Stable branch: `1.$MINOR.x`, where `$MINOR` is current minor version (e.g. `1.1.x` during 1.1.x series)
 
-	Note that sbt is always built with the previous stable release.  For example, the [1.0.x](https://github.com/sbt/sbt/tree/1.0.x) branch is built with 0.13.13 and the [v0.13.13](https://github.com/sbt/sbt/tree/v0.13.13) tag is built with 0.13.12.
+If you're working on a bug fix, it's a good idea to start with the `1.$MINOR.x` branch. Since we can always safely merge from stable to `1.x`, but not other way around.
 
-4. To build the launcher and publish all components locally,
+### Instruction to build all modules from source
 
-		$ sbt
-		> publishLocal
+1. Install the current stable binary release of sbt (see [Setup]), which will be used to build sbt from source.
+2. Get the source code.
 
-5. To use this locally built version of sbt, copy your stable `~/bin/sbt` script to `~/bin/xsbt` and change it to use the launcher jar at `<sbt>/launch/target/sbt-launch.jar`.
+   ```
+   $ mkdir sbt-modules
+   $ cd sbt-modules
+   $ for i in sbt io util librarymanagement zinc; do \
+     git clone https://github.com/sbt/$i.git && (cd $i; git checkout -b 1.1.x origin/1.1.x)
+   done
+   $ cd sbt
+   $ ./sbt-allsources.sh
+   ```
 
-	Directory `target` is removed by `clean` command. Second solution is using the artifact stored in the local ivy repository.
+3. To build the launcher and publish all components locally,
 
-	The launcher is located in:
+   ```
+   $ ./sbt-allsources.sh
+   > ;{../io}/publishLocal; {../util}/publishLocal; {../librarymanagement}/publishLocal; {../zinc}/publishLocal
+   > publishLocal
+   ```
 
-    $HOME/.ivy2/local/org.scala-sbt/sbt-launch/0.13.9/jars/sbt-launch.jar
+### Instruction to build just sbt
 
-	for v0.13.9 tag, or in:
+If the change you are making is contained in sbt/sbt, you could publishLocal on sbt/sbt:
 
-    $HOME/.ivy2/local/org.scala-sbt/sbt-launch/0.13.10-SNAPSHOT/jars/sbt-launch.jar
+```
+$ sbt
+> publishLocal
+```
 
-	for the development branch.
+### Using the locally built sbt
 
-## Modifying sbt
+To use the locally built sbt, set the version in `build.properties` file to `1.$MINOR.$PATCH-SNAPSHOT`, or pass it in as `-Dsbt-version` property.
 
-1. When developing sbt itself, run `compile` when checking compilation only.
+```
+$ cd ../hello
+$ sbt -Dsbt-version=1.1.2-SNAPSHOT
+```
```

Comment:
Does this method differ from `sbt -sbt-version`? I was a bit confused by the subsequent message about the mismatch between the `build.properties` version and the active version which talked about the `reboot` command. So I went back to editing `build.properties.`. Perhaps just suggest that way of doing it (which should be familiar to people)

Reply:
Yea. Let's keep it simple.